zephyr/lc3_decoder.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022 Bose Corporation All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIB_LIBLC3CODEC_ZEPHYR_LC3_DECODER_H_
+#define LIB_LIBLC3CODEC_ZEPHYR_LC3_DECODER_H_
+
+/* This file is a c-wrapper for the c++ LC3 lib. Please refer to the C++ API documentation
+ * for a full description of the API.
+ * General usage:
+ * 1) Call _create to allocate memory for the c++ object using the c++-lib allocation mechanism.
+ * 2) Use the wrapper functions as needed passing the allocated object as first parameter.
+ * 3) After usage call _destroy to free up the memory.
+ *    Invalid usage of the object reference after a call to _destroy cannot be detected and will
+ *    most likely result in a crash. */
+
+#include <stdint.h>
+#include "lc3_config.h"
+
+#ifdef __cplusplus
+#define LC3_EXTERN_C extern "C"
+#else
+#define LC3_EXTERN_C
+#endif
+
+typedef void* Lc3Decoder_t;
+
+#define LC3_DECODE_ERROR_FREE                       0x00
+#define LC3_DECODE_INVALID_CONFIGURATION            0x01
+#define LC3_DECODE_INVALID_BYTE_COUNT               0x02
+#define LC3_DECODE_INVALID_X_OUT_SIZE               0x03
+#define LC3_DECODE_INVALID_BITS_PER_AUDIO_SAMPLE    0x04
+#define LC3_DECODE_DECODER_ALLOCATION_ERROR         0x05
+
+/**
+ * For detailed description please see underlying LC3 codec implementation.
+ * @param Fs_ Sample frequency HZ
+ * @param N_ms_ Frame duration
+ * @return the Decoder instance
+ */
+LC3_EXTERN_C Lc3Decoder_t Lc3Decoder_create(uint16_t Fs_, lc3_frame_duration_t N_ms_);
+LC3_EXTERN_C void Lc3Decoder_destroy(Lc3Decoder_t inst_ptr);
+
+/**
+ * Decode a single frame of 16 bit audio for one channel.
```

Comment:
Could we make a function that decodes an entire stream (1 or more channels) at a time? Perhaps later?

Reply:
That would require a streaming interface to the ISO channel in zephyr as well as a way to configure the audio output configuration (for example if samples should be interleaved - currently the interface provides a callback delivering one ISO packet at the time - typically one audio frame - potentially with multiple channels. If we need more wrapping around the LC3 lib I think it needs to be in the Zephyr project and not the LC3 project.
If you refer to multiple channels for a single audio frame - then yes we could, but the output format is not documented nor configurable, hence I avoided exposing that functionality until someone did some analysis of the code and updated the documentation:-)

Reply:
Sorry, I think I had mixed up the terms of `packet` and `frame` here. 

![image](https://user-images.githubusercontent.com/2520413/149326524-9ab7f12e-b85c-4006-b3e5-39acf2bc8ec5.png)

So a frame is for a single channel, and there are 1 or more frames in a packet. 

This is fine then :)